test/http-api/test-pin.js

```diff
@@ -0,0 +1,155 @@
+        it('pins object directly if specified', (done) => {
+          api.inject({
+            method: 'POST',
+            url: `/api/v0/pin/add?arg=${keys.leaf}&recursive=false`
+          }, (res) => {
+            expect(res.statusCode).to.equal(200)
+            expect(res.result[0].hash).to.equal(keys.leaf)
+            expect(res.result[0].type).to.equal('direct')
+            done()
+          })
+        })
+      })
+
+      describe('/pin/ls (with path)', () => {
+        it('finds specified pinned object', (done) => {
+          api.inject({
+            method: 'GET',
+            url: `/api/v0/pin/ls?arg=/ipfs/${keys.root}/branch/subLeaf`
+          }, (res) => {
+            expect(res.statusCode).to.equal(200)
+            expect(res.result[0].hash).to.equal(keys.subLeaf)
+            expect(res.result[0].type).to.equal('indirect')
+            expect(res.result[0].through).to.equal(keys.root)
+            done()
+          })
+        })
+      })
+
+      describe('/pin/ls (without path or type)', () => {
+        it('finds all pinned objects', (done) => {
+          api.inject({
+            method: 'GET',
+            url: ('/api/v0/pin/ls')
+          }, (res) => {
+            expect(res.statusCode).to.equal(200)
+            expect(res.result.length).to.equal(4)
+            expect(res.result).to.include({type: 'direct', hash: keys.leaf})
+            expect(res.result).to.include({type: 'recursive', hash: keys.root})
+            expect(res.result).to.include({type: 'indirect', hash: keys.branch})
+            expect(res.result).to.include({type: 'indirect', hash: keys.subLeaf})
+            done()
+          })
+        })
+      })
+
+      describe('/pin/rm (direct)', () => {
+        it('unpins only directly pinned objects if specified', (done) => {
+          api.inject({
+            method: 'POST',
+            url: `/api/v0/pin/rm?arg=${keys.leaf}&recursive=false`
+          }, (res) => {
+            expect(res.statusCode).to.equal(200)
+            expect(res.result[0]).to.equal(keys.leaf)
+
+            api.inject({
+              method: 'POST',
+              url: `/api/v0/pin/rm?arg=${keys.root}&recursive=false`
+            }, (res) => {
+              expect(res.statusCode).to.equal(500)
+              expect(res.result.Message).to.equal(
+                'Failed to remove pin: ' +
+                'QmWQwS2Xh1SFGMPzUVYQ52b7RC7fTfiaPHm3ZyTRZuHmer ' +
+                'is pinned recursively'
+              )
+              done()
+            })
+          })
+        })
+      })
+
+      describe('/pin/rm', () => {
+        it('unpins recursively by default', (done) => {
+          api.inject({
+            method: 'POST',
+            url: `/api/v0/pin/rm?arg=${keys.root}`
+          }, (res) => {
+            expect(res.statusCode).to.equal(200)
+            expect(res.result[0]).to.equal(keys.root)
+            done()
+          })
+        })
+      })
+    })
```

Comment:
I see the tests here for the pin routes, yet the endpoint loading is commented out.